File: widgetssdk/src/main/java/com/glia/widgets/core/fileupload/domain/RemoveFileAttachmentUseCase.java
package com.glia.widgets.core.fileupload.domain;

import com.glia.widgets.core.fileupload.FileAttachmentRepository;
import com.glia.widgets.core.fileupload.model.FileAttachment;

public class RemoveFileAttachmentUseCase {
    private final FileAttachmentRepository repository;

    public RemoveFileAttachmentUseCase(FileAttachmentRepository repository) {
        this.repository = repository;
    }

    public void execute(FileAttachment attachment) {
        repository.detachFile(attachment);
    }
}

File: widgetssdk/src/main/java/com/glia/widgets/core/fileupload/domain/RemoveFileAttachmentUseCase.kt
package com.glia.widgets.core.fileupload.domain

import com.glia.widgets.core.fileupload.EngagementFileAttachmentRepository
import com.glia.widgets.core.fileupload.model.LocalAttachment

internal class RemoveFileAttachmentUseCase(private val fileAttachmentUseCase: ChatFileAttachmentRepositoryUseCase) {
    operator fun invoke(attachment: LocalAttachment?) {
        fileAttachmentUseCase().detachFile(attachment)
    }
}
